tests/core/input/test_input_event_key.h

```diff
@@ -198,141 +209,155 @@ TEST_CASE("[InputEventKey] Keys are correctly matched based on action") {
 	bool pressed = false;
 	float strength, raw_strength = 0.0;
 
-	InputEventKey key;
+	Ref<InputEventKey> key;
+	key.instantiate();
 
 	// Nullptr.
-	CHECK_MESSAGE(key.action_match(nullptr, false, 0.0f, &pressed, &strength, &raw_strength) == false, "nullptr as key reference should result in false");
+	CHECK_MESSAGE(key->action_match(nullptr, false, 0.0f, &pressed, &strength, &raw_strength) == false, "nullptr as key reference should result in false");
 
 	// Match on keycode.
-	key.set_keycode(Key::SPACE);
-	Ref<InputEventKey> match = key.create_reference(Key::SPACE);
-	Ref<InputEventKey> no_match = key.create_reference(Key::ENTER);
+	key->set_keycode(Key::SPACE);
+	Ref<InputEventKey> match = InputEventKey::create_reference(Key::SPACE);
+	Ref<InputEventKey> no_match = InputEventKey::create_reference(Key::ENTER);
```

Comment:
These too are possibly strange as they might depend on the assumption that `create_reference` isn't a static method, can rework these tests if needed or remove parts that are unnecessary, but just fixed some of the weirdness of them now